code/modules/food&drinks/food/snacks_pie.dm

```diff
@@ -27,7 +27,42 @@
 		var/turf/T = get_turf(hit_atom)
 		new/obj/effect/decal/cleanable/pie_smudge(T)
 		reagents.reaction(hit_atom, TOUCH)
-		qdel(src)
+
+		if(!ismob(hit_atom)) //if the creampie misses
+			qdel(src)
+			return
+
+		else //if the creampie hits
+			var/mob/living/H = hit_atom
+			var/image/creamoverlay = image('icons/effects/creampie.dmi')
+
+			if(istype(H, /mob/living/carbon/human))
+				var/mob/living/carbon/human/S = hit_atom
+				if(S.dna && S.dna.species && ("tail_lizard" in S.dna.species.mutant_bodyparts))
+					creamoverlay.icon_state = "creampie_lizard"
+				else creamoverlay.icon_state = "creampie_human"
+			else if(istype(H, /mob/living/carbon/monkey))
+				creamoverlay.icon_state = "creampie_monkey"
+			else if(istype(H, /mob/living/carbon/alien))
+				creamoverlay.icon_state = "creampie_xenomorph"
+				if(istype(H, /mob/living/carbon/alien/humanoid/royal)) //lets not mess with queen yet
+					return
+			else if(istype(H, /mob/living/silicon/ai))
+				creamoverlay.icon_state = "creampie_ai"
+			else if(istype(H, /mob/living/simple_animal/drone))
+				creamoverlay.icon_state = "creampie_drone"
+			else if(istype(H, /mob/living/simple_animal/pet/dog/corgi))
+				creamoverlay.icon_state = "creampie_corgi"
+			else return //if the target mob isn't compatible
+
+			if(istype(H, /mob/living/carbon/human))
+				H.Weaken(1) //splat!
+				H.adjust_blurriness(1)
+			visible_message("<span class='userdanger'>[H] was creamed by the [src]!!</span>")
+			H.overlays += creamoverlay
+
+		update_icon()
```

Comment:
what's the point of updating the icon when we're qdeling the item right after that?